Development.md

```diff
@@ -0,0 +1,273 @@
+* Release artifacts to enable `convox rack update`
+
+## Sponsorship and Mentoring
+
+Developing Rack will incur AWS costs. If this is an obstacle for you to contribute you can contact [support@convox.com](mailto:support@convox.com) to request sponsorship.
+
+Much of the technical design and implementation in Rack requires understanding of AWS, Docker, Golang, systems engineering and more. If you would like to better learn these systems to contribute, you can contact support@convox.com, join the [Public Slack](http://invite.convox.com/), or open up issues on [GitHub](http://github.com/convox/rack) to ask questions and/or request a mentor.
+
+## Sandbox AWS Rack Install
+
+Rack consumes numerous AWS and Docker APIs. The easiest way to develop Rack is with real AWS access keys interacting with real AWS resources like a Dynamo Table, ECS Cluster, and CloudFormation Stack.
+
+This is easy to bootstrap with the Rack project itself:
+
+```
+$ convox install --stack-name dev
+```
+
+You can also use any existing Rack with the caveat that running a laptop Rack against it could have side effects like terminating instances.
+
+## AWS Rack Ingress
+
+Parts of the API like `convox ps --stats` interact with the Docker daemon running on every AWS instance. To enable `convox ps --stats` to work from a laptop you will want to open up access to the instance Docker daemons:
+
+* Open the [Security Group Management Console](https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#SecurityGroups)
+* Select the Security Group with the Group Name like "dev-SecurityGroup-4PNOYR5HUH83" and the Description "Instances"
+* Click the Inbound tab, then the Edit button, then the Add Rule button
+* Keep Custom TCP Rule and TCP protocol
+* Add "2376" for Port Range, and select "My IP" for Source
+* Click the Save button
+
+Warning: Do not expose instance port 2376 to "Anywhere"! This will expose your Docker daemons to the whole Internet.
+
+## Rack Golang Project
+
+Rack is written in Golang. To setup a Go environment, see the excellent [Getting Started](https://golang.org/doc/install) docs. You can then clone and build the project with the `go get` tool:
+
+```
+$ go get github.com/convox/rack/...
+```
+
+After this, `which convox` should refer to `$GOPATH/bin/convox`.
+
+## Laptop Rack Environment
+
+The laptop Rack is running an API process that has AWS Access Keys, AWS resource names, and other various settings in its environment. You need to copy this to your laptop
+
+```
+$ cd $GOPATH/src/github.com/convox/rack
+
+# Introspect the dev rack to find the PID of the API web process
+
+$ STACK_NAME=$(convox api get /system | jq -r .name)
+$ WEB_PID=$(convox api get /apps/$STACK_NAME/processes | jq -r '.[] | select(.name == "web") | .id' | head -1)
+
+# Introspect the API web process to get its environment
+
+$ convox exec $WEB_PID env --app $STACK_NAME > .env
+```
+
+Now you have a bunch of secrets that will let your laptop interact with AWS APIs:
```

Comment:
It's about the developer not the laptop:) -  "Now you have a bunch of secrets that will let you interact with AWS APIs from your laptop:"